ivadomed/loader/mri3d_subvolume_segmentation_dataset.py

```diff
@@ -35,24 +40,31 @@ class MRI3DSubVolumeSegmentationDataset(Dataset):
     """
 
     def __init__(self, filename_pairs, transform=None, length=(64, 64, 64), stride=(0, 0, 0), slice_axis=0,
-                 task="segmentation", soft_gt=False, is_input_dropout=False):
+                 task="segmentation", soft_gt=False, is_input_dropout=False, disk_cache=None):
         self.filename_pairs = filename_pairs
-        self.handlers = []
-        self.indexes = []
+
+        # could be a list of tuple of objects OR path objects to the actual disk equivalent.
+        # behaves differently depend on if self.cache is set to true or not.
+        self.handlers: List[tuple] = []
+
+        self.indexes: list = []
         self.length = length
         self.stride = stride
         self.prepro_transforms, self.transform = transform
         self.slice_axis = slice_axis
-        self.has_bounding_box = True
+        self.has_bounding_box: bool = True
         self.task = task
         self.soft_gt = soft_gt
         self.is_input_dropout = is_input_dropout
+        self.disk_cache: bool = disk_cache
 
         self._load_filenames()
         self._prepare_indices()
 
     def _load_filenames(self):
         """Load preprocessed pair data (input and gt) in handler."""
+        path_cache: Path = Path(create_temp_directory())
```

Comment:
In 2D this was defined as `path_temp`. I think it would be better to have the same name for consistency